sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -182,6 +194,47 @@ case class GlobalLimitExec(limit: Int, child: SparkPlan) extends BaseLimitExec {
     copy(child = newChild)
 }
 
+/**
+ * Skip the first `offset` elements then take the first `limit` of the following elements in
+ * the child's single output partition.
+ */
+case class GlobalLimitAndOffsetExec(
+    limit: Int,
+    offset: Int,
+    child: SparkPlan) extends BaseLimitExec {
+
+  override def requiredChildDistribution: List[Distribution] = AllTuples :: Nil
+
+  override def outputPartitioning: Partitioning = child.outputPartitioning
+
+  override def outputOrdering: Seq[SortOrder] = child.outputOrdering
```

Comment:
We need to override the above 2 methods, as `BaseLimitExec` already defines them

Reply:
It seems the base class changed before I reopen this PR.